identity-service/src/routes/reactions.js

```diff
@@ -0,0 +1,79 @@
+const { handleResponse, errorResponseBadRequest, errorResponseServerError, successResponse } = require('../apiHelpers')
+const authMiddleware = require('../authMiddleware')
+const models = require('../models')
+const { logger } = require('../logging')
+
+const MAX_REACTIONS_PER_FETCH = 100
+
+const handleReaction = async ({ senderWallet, entityType, entityId, libs, reaction }) => {
+  const { solanaWeb3Manager } = libs
+  const currentSlot = await solanaWeb3Manager.getSlot()
+
+  // Unique constraint on entityId + senderWallet
+  // prevents a user from reacting to the same entity multiple times
+  const now = Date.now()
+  await models.Reactions.create({
+    slot: currentSlot,
+    reaction,
+    senderWallet,
+    entityId,
+    entityType,
+    createdAt: now,
+    updatedAt: now
+  })
+}
+
+const getReactions = async ({ startIndex, limit }) => {
+  const reactions = await models.Reactions.findAll({
+    where: {
+      id: { [models.Sequelize.Op.gte]: startIndex }
+    },
+    order: [[models.Sequelize.col('id'), 'ASC']],
+    limit
+  })
+  return reactions
+}
+
+module.exports = function (app) {
+  /**
+   * POST a new reaction, represented by a numeric ID (reaction) and entityId (entity being reacted upon)
+   */
+  app.post('/reactions', authMiddleware, handleResponse(async (req, res, next) => {
+    // Validation
+    const senderWallet = req.user.walletAddress
+    if (!senderWallet) return errorResponseBadRequest('Did not pass a valid user id')
+    const { entityId, reaction } = req.body
+    if (!entityId || !reaction) return errorResponseBadRequest('Did not pass a valid reaction or entityId')
+    const parsedReaction = parseInt(reaction)
```

Comment:
super nit, but maybe space after if to indicate these destructure + if checks are kinda a block?

Reply:
agree